go-controller/pkg/node/secondary_node_network_controller_test.go

```diff
@@ -268,9 +277,10 @@ var _ = Describe("SecondaryNodeNetworkController", func() {
 
 		By("creating secondary network controller for user defined primary network")
 		cnnci := CommonNodeNetworkControllerInfo{name: nodeName, watchFactory: &factoryMock}
-		controller, err := NewSecondaryNodeNetworkController(&cnnci, NetInfo, vrf)
+		controller, err := NewSecondaryNodeNetworkController(&cnnci, NetInfo, vrf, ipRulesManager)
 		Expect(err).NotTo(HaveOccurred())
 		Expect(controller.gateway).To(Not(BeNil()))
+		Expect(controller.gateway.ruleManager).To(Not(BeNil()))
```

Comment:
would be good to check if ip rules created/deleted for MasqIP on the testNS for controller start and cleanup.

Reply:
yeah I should probably add that. tests take soo much time hehe

Reply:
yes done now!